src/executor.cc

```diff
@@ -99,7 +99,16 @@ Result Executor::Execute(Engine* engine,
     if (!r.IsSuccess())
       return r;
 
-    buf->SetData(info.values);
+    std::vector<uint8_t>* data = buf->ValuePtr();
+
+    data->clear();
```

Comment:
This seems like a separate fix?

Reply:
It's just a more general solution that doesn't assume the destination buffer is in uint8 format. It works both with the png images and buffers of any format. Basically we keep the destination buffer format but directly copy the bytes and then adjust the element size of the destination buffer based on the format.